src/report.rs

```diff
@@ -0,0 +1,256 @@
+//! Optionally writes an html-formatted summary report after running a load test.
+
+use crate::metrics;
+
+use std::collections::BTreeMap;
+use std::mem;
+
+use serde::Serialize;
+
+/// Defines the metrics reported about requests.
+#[derive(Debug, Clone, Serialize)]
+pub struct RequestMetric {
+    pub method: String,
+    pub name: String,
+    pub number_of_requests: usize,
+    pub number_of_failures: usize,
+    pub response_time_average: String,
+    pub response_time_minimum: usize,
+    pub response_time_maximum: usize,
+    pub requests_per_second: String,
+    pub failures_per_second: String,
+}
+
+/// Defines the metrics reported about responses.
+#[derive(Debug, Clone, Serialize)]
+pub struct ResponseMetric {
+    pub method: String,
+    pub name: String,
+    pub percentile_50: String,
+    pub percentile_60: String,
+    pub percentile_70: String,
+    pub percentile_80: String,
+    pub percentile_90: String,
+    pub percentile_95: String,
+    pub percentile_99: String,
+    pub percentile_100: String,
+}
+
+/// Helper to generate a single response metric.
+pub fn get_response_metric(
```

Comment:
I like this function, but I'm open to better ways to do this. I originally was doing all this work each time directly, and it was a lot of code duplication. It's perhaps a little odd how I build a Vector and then `take` the Strings out of the vector, but it saves having to do it more manually for each column.